packages/new-deploy/scripts/verify.ts

```diff
@@ -0,0 +1,245 @@
+    optimizations = 999999,
+  ) {
+    this._data = v;
+    this.chainId = chainId;
+    this.compiler = compiler;
+    this.optimizations = optimizations;
+  }
+
+  get name(): string {
+    return this._data.name;
+  }
+
+  get specifier(): string {
+    return this._data.specifier;
+  }
+
+  get address(): string {
+    return this._data.address;
+  }
+
+  get constructorArguments(): ReadonlyArray<unknown> | undefined {
+    return this._data.constructorArguments;
+  }
+
+  get encodedConstructorArguments(): string | undefined {
+    return this._data.encodedConstructorArguments;
+  }
+
+  get GUID(): string | undefined {
+    return this._data.GUID;
+  }
+
+  set GUID(GUID: string | undefined) {
+    this._data.GUID = GUID;
+  }
+
+  get verified(): boolean {
+    return this._data.verified ?? false;
+  }
+
+  set verified(v: boolean) {
+    this._data.verified = v;
+  }
+
+  /// Extract the foundry profile from a specifier.
+  /// This is brittle code as it relies on the contents of `foundry.toml` not
+  /// changing, as well as on our repo layout
+  get profile(): string {
+    if (this.specifier.indexOf('core') !== -1) return 'core';
+    if (this.specifier.indexOf('bridge') !== -1) return 'bridge';
+    if (this.specifier.indexOf('router') !== -1) return 'router';
+    throw new Error(`bad specifier: ${this.specifier}`);
+  }
+
+  get argsFlag(): string | undefined {
+    if (!this.encodedConstructorArguments) return;
+    return `--constructor-args=${this.encodedConstructorArguments}`;
+  }
+
+  get verifyCommand(): string {
+    if (this._data.verifyCommand) return this._data.verifyCommand;
+
+    this._data.verifyCommand = forgeVerifyCommand(this);
+    return this._data.verifyCommand;
+  }
+
+  get checkCommand(): string {
+    if (!this.GUID) throw new Error('No GUID. Call `verify` first');
+    return forgeCheckCommand(this);
+  }
+
+  async verify(): Promise<void> {
+    if (this.verified) {
+      console.log(this.name, 'Already Verified');
+      return;
+    }
+
+    const toRun = this.verifyCommand;
+    try {
+      console.log(`verifying ${this.name}@${this.address}`);
+      const res = await execAsync(toRun);
+      this.GUID = extractGuidFromFoundry(res);
+      if (!this.GUID) this.verified = true;
+    } catch (e: unknown) {
+      const err = `${e}`;
+      if (err.indexOf('Contract source code already verified') === -1) {
+        console.error({ toRun });
+        throw e;
+      }
+
+      // only "already verified" errors will reach this line
+      this.verified = true;
+    }
+  }
+
+  /// returns the error string
+  async check(): Promise<string | undefined> {
+    if (!this.GUID) return;
+    const toRun = this.checkCommand;
+    const { stderr } = await execAsync(toRun);
+    if (stderr.indexOf('NOTOK') !== -1) {
+      return stderr.split('`')[4];
+    } else {
+      this.verified = true;
+    }
+  }
+
+  toJSON(): VerificationOutput {
+    return this._data;
+  }
+}
+
+async function run() {
```

Comment:
hmmm in my mind, ideally we could split this up into a diff file than the class?

Reply:
the future change should be to incorporate the verification class into the deploy script. instead of doing the intermediate change, I'd like to just leave it how it is until we do the proper fix. a mono-file script is fine if messy.